src/services/auth.service.js

```diff
@@ -5,27 +5,40 @@ const mongoose = require('mongoose')
 mongoose.Promise = Promise
 
 const User = require('../models/user.model')
-const tokenLife = config.TOKEN_LIFE
+const tokenExpTime = config.TOKEN_LIFE
+const moment = require('moment')
 const _ = require('lodash')
 
 const randomBytes = Promise.promisify(require('crypto').randomBytes)
 
 module.exports = () => {
   const generateToken = user =>
     jwt.sign(user, config.SECRET, {
-      expiresIn: parseInt(tokenLife)
+      expiresIn: parseInt(tokenExpTime)
     })
 
   const getToken = user => `JWT ${generateToken(user)}`
 
+  const setExpirationDate = () =>
+    moment().add((parseInt(tokenExpTime) / 3600), 'h')
+
+  const refreshToken = user =>
+    Promise.all([
+      getToken(_.omit(user, ['exp'])),
+      setExpirationDate()
+    ])
+    .then(([token, expiresIn]) =>
+      JSON.stringify({ token, expiresIn }))
+
   const resetPasswordToken = user => {
     const date = new Date()
     user = _.extend(user, User)
-    return randomBytes(parseInt(config.RANDOM_BYTES)).then(buffer => {
-      user.resetPasswordToken = buffer.toString('hex')
-      user.resetPasswordExpires = date.setSeconds(date.getSeconds() + parseInt(tokenLife))
-      return user.save()
-    })
+    return randomBytes(parseInt(config.RANDOM_BYTES))
+      .then(buffer => {
+        user.resetPasswordToken = buffer.toString('hex')
```

Comment:
This method is used to encode each byte of the buffer as two hexadecimal characters.